src/Compilers/CSharp/Portable/Syntax/InterceptableLocation.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.Versioning;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.CSharp.Syntax
+{
+    /// <summary>
+    /// Contains values suitable for populating System.Runtime.CompilerServices.InterceptsLocationAttribute for a given call.
+    /// </summary>
+    [RequiresPreviewFeatures]
```

Comment:
> [RequiresPreviewFeatures] [](http://example.com/codeflow?start=3&length=26)

Where this attribute is enforced?

Reply:
An analyzer which ships in the dotnet SDK enforces it.